src/Tribe/Editor/Blocks/Abstract.php

```diff
@@ -172,14 +173,46 @@ public function load() {
 	 * @return bool Whether the post has this block.
 	 */
 	public function has_block( $post = null ) {
-		if ( ! is_numeric( $post ) ) {
+		$wp_post = null;
+		$post_id = null;
+
+		if ( is_numeric( $post ) || $post === null ) {
 			$wp_post = get_post( $post );
-			if ( $wp_post instanceof WP_Post ) {
-				$post = $wp_post->post_content;
-			}
+		} elseif ( $post instanceof WP_Post ) {
+			$wp_post = $post;
+		}
+
+		if ( $wp_post instanceof WP_Post ) {
+			$post    = $wp_post->post_content;
+			$post_id = $wp_post->ID;
 		}
 
-		return false !== strpos( (string) $post, '<!-- wp:' . $this->name() );
+		$has_block = false !== strpos( (string) $post, '<!-- wp:' . $this->name() );
+
+		/**
+		 * Filters whether the post has this block.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $has_block Whether the post has this block.
+		 * @param WP_Post|null $wp_post The post object.
+		 * @param int|null $post_id The post ID.
+		 * @param string $block_name The block name.
+		 * @param Tribe__Editor__Blocks__Abstract $this The block object.
+		 */
+		$has_block = (bool) apply_filters( 'tec_block_has_block', $has_block, $wp_post, $post_id, $this->name(), $this );
+
+		/**
+		 * Filters whether the post has this block.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $has_block Whether the post has this block.
+		 * @param WP_Post|null $wp_post The post object.
+		 * @param int|null $post_id The post ID.
+		 * @param Tribe__Editor__Blocks__Abstract $this The block object.
+		 */
+		return (bool) apply_filters( 'tec_block_' . $this->name() . '_has_block', $has_block, $wp_post, $post_id, $this );
```

Comment:
```suggestion
		$block_name =$this->name();

		/**
		 * Filters whether the post has this block.
		 *
		 * @since TBD
		 *
		 * @param bool $has_block Whether the post has this block.
		 * @param WP_Post|null $wp_post The post object.
		 * @param int|null $post_id The post ID.
		 * @param Tribe__Editor__Blocks__Abstract $this The block object.
		 */
		return (bool) apply_filters( "tec_block_{$block_name}_has_block", $has_block, $wp_post, $post_id, $this );
```